typespec-tests/src/main/java/com/cadl/multipleapiversion/NoApiVersionClient.java

```diff
@@ -0,0 +1,118 @@
+public final class NoApiVersionClient {
+    @Generated
+    private final NoApiVersionClientImpl serviceClient;
+
+    /**
+     * Initializes an instance of NoApiVersionClient class.
+     * 
+     * @param serviceClient the service client implementation.
+     */
+    @Generated
+    NoApiVersionClient(NoApiVersionClientImpl serviceClient) {
+        this.serviceClient = serviceClient;
+    }
+
+    /**
+     * The action operation.
+     * <p>
+     * <strong>Query Parameters</strong>
+     * </p>
+     * <table border="1">
+     * <caption>Query Parameters</caption>
+     * <tr>
+     * <th>Name</th>
+     * <th>Type</th>
+     * <th>Required</th>
+     * <th>Description</th>
+     * </tr>
+     * <tr>
+     * <td>param1</td>
+     * <td>String</td>
+     * <td>No</td>
+     * <td>A sequence of textual characters.</td>
+     * </tr>
+     * </table>
+     * You can add these to a request with {@link RequestOptions#addQueryParam}
+     * 
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return the {@link Response}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<Void> actionWithResponse(RequestOptions requestOptions) {
+        return this.serviceClient.actionWithResponse(requestOptions);
+    }
+
+    /**
+     * The action operation.
+     * 
+     * @param param1 A sequence of textual characters.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void action(String param1) {
+        // Generated convenience method for actionWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        if (!Arrays.asList("2022-12-01-preview").contains(serviceClient.getServiceVersion().getVersion())) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException("Parameter param1 is only available in api-version 2022-12-01-preview."));
+        }
```

Comment:
We actually still need to use the ServiceVersion when doing versioning resiliency.